airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -87,7 +87,17 @@ def get_link(self, operator: BaseOperator, dttm: datetime):
         return BIGQUERY_JOB_DETAILS_LINK_FMT.format(job_id=job_id)
 
 
-class BigQueryCheckOperator(CheckOperator):
+class BigQueryDbHookMixin:
+    def get_db_hook(self) -> BigQueryHook:
+        return BigQueryHook(
+            gcp_conn_id=self.gcp_conn_id,
+            use_legacy_sql=self.use_legacy_sql,
+            location=self.location,
+            impersonation_chain=self.impersonation_chain,
+        )
+
+
+class BigQueryCheckOperator(CheckOperator, BigQueryDbHookMixin):
```

Comment:
This is not a pattern recommended by Google:
https://s.apache.org/google-operators

It is rather preferred to instantiated the hook in execute method in operator. In this way refactoring and future changes are simpler and we reduce cross dependencies between classes.